File: model/src/main/java/io/sparkled/model/entity/SequenceStatus.java
package io.sparkled.model.entity;

public enum SequenceStatus {
    /**
     * Has no animation data whatsoever, cannot be played.
     */
    NEW,

    /**
     * Has unpublished animation data, can be played with the previously published data.
     */
    DRAFT,

    /**
     * Has up-to-date animation data, can be played.
     */
    PUBLISHED
}

File: model/src/main/kotlin/io/sparkled/model/entity/SequenceStatus.kt
package io.sparkled.model.entity

import com.fasterxml.jackson.annotation.JsonEnumDefaultValue

enum class SequenceStatus {
    /**
     * Has no animation data, cannot be played.
     */
    @JsonEnumDefaultValue
    NEW,

    /**
     * Has unpublished animation data, can be played with the previously published data.
     */
    DRAFT,

    /**
     * Has up-to-date animation data, can be played.
     */
    PUBLISHED
}
